common.go

```diff
@@ -43,15 +45,16 @@ func retryWithLimitedNumOfRetries(targetFunc retryableFunc, numOfRetries int, de
 	var err error
 	var continueRetrying bool
 	for retryNo <= numOfRetries {
-		time.Sleep(delay) //delay between retries
+		retryNo++
+		time.Sleep(delay * time.Duration(retryNo)) //delay between retries
```

Comment:
Is it meant to increase the delay interval in each retry? The delay interval is 1 sec in the 1st time, 2 secs in 2nd time, so on and so far.

Reply:
yes, exactly. you can read more about the advantage of this here https://dzone.com/articles/understanding-retry-pattern-with-exponential-back

Reply:
Thanks!